xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/bridge/event/DocumentDeletedEvent.java

```diff
@@ -42,28 +42,34 @@ public class DocumentDeletedEvent extends AbstractDocumentEvent
      * changes.
      */
     private static final long serialVersionUID = 1L;
-    
+
+    /**
+     * The reference of the deleted document.
+     */
+    private final DocumentReference sourceReference;
+
     /**
      * Constructor initializing the event filter with an
      * {@link org.xwiki.observation.event.filter.AlwaysMatchingEventFilter}, meaning that this event will match any
      * other document delete event.
      */
     public DocumentDeletedEvent()
     {
-        
+        this.sourceReference = null;
     }
-    
+
     /**
      * Constructor initializing the event filter with a {@link org.xwiki.observation.event.filter.FixedNameEventFilter},
      * meaning that this event will match only delete events affecting the same document.
      * 
      * @param documentReference the reference of the document to match
      */
-    public DocumentDeletedEvent(DocumentReference documentReference)    
+    public DocumentDeletedEvent(DocumentReference documentReference)
     {
         super(documentReference);
+        this.sourceReference = documentReference;
```

Comment:
I think this need some comments since it shouldn't be necessary normally since the ref is passed in the parent constructor and it's supposed to offer a way to get it (I've noticed that it's serialized though which seems weird and would need to be checked for accuracy).